lib/chef/resource/windows_package.rb

```diff
@@ -69,10 +69,26 @@ def source(arg=nil)
           @source
         else
           raise ArgumentError, "Bad type for WindowsPackage resource, use a String" unless arg.is_a?(String)
-          Chef::Log.debug("#{package_name}: sanitizing source path '#{arg}'")
-          @source = ::File.absolute_path(arg).gsub(::File::SEPARATOR, ::File::ALT_SEPARATOR)
+          if is_url?(arg)
+            @source = arg
+          else
+            @source = ::File.absolute_path(arg).gsub(::File::SEPARATOR, ::File::ALT_SEPARATOR)
+          end
         end
       end
+
+      private
+
+      def is_url?(source)
```

Comment:
Whatever we do, this makes three places we have this method (this resource/provider, plus rpm provider), it should probably be a mixin.